src/resumable-subscription.ts

```diff
@@ -1,61 +1,31 @@
 import { TokenProvider } from './token-provider';
-import { Subscription } from './subscription';
-import { ErrorResponse, Event } from './base-client';
+import { ErrorResponse } from './base-client';
 import { RetryStrategy, ExponentialBackoffRetryStrategy, Retry, DoNotRetry } from './retry-strategy';
 import { Logger } from './logger';
+import { BaseSubscription, SubscribeOptions, SubscriptionEvent } from './base-subscription' 
 
-export interface ResumableSubscribeOptions {
-    path: string;
+export interface ResumableSubscribeOptions extends SubscribeOptions {
     lastEventId?: string;
-    tokenProvider?: TokenProvider;
-    onOpening?: () => void;
-    onOpen?: () => void;
-    onEvent?: (event: Event) => void;
-    onEnd?: () => void;
-    onError?: (error: Error) => void;
     retryStrategy?: RetryStrategy;
-    onRetry?: ()=>void;
-    logger?: Logger;
+    tokenProvider?: TokenProvider; 
 }
 
-export enum ResumableSubscriptionState {
-    UNOPENED = 0,
-    OPENING,      // can be visited multiple times
-    OPEN,         // called onOpen(); expecting message
-    ENDING,       // received EOS message; response not yet finished
-    ENDED         // called onEnd() or onError(err)
-}
-
-// Asserts that the subscription state is one of the specified values,
-// otherwise logs the current value.
-export function assertState(stateEnum, states = []) {
-    const check = states.some(state => stateEnum[state] === this.state);
-    const expected = states.join(', ');
-    const actual = stateEnum[this.state];
-    console.assert(check, `Expected this.state to be ${expected} but it is ${actual}`);
-    if (!check) {
-        console.trace();
-    }
-};
-
-// pattern of callbacks: ((onOpening (onOpen onEvent*)?)? (onError|onEnd)) | onError
 export class ResumableSubscription {
```

Comment:
I think it would make sense - if we were writing prose. 

In our case I think it's better to keep the producer/consumer terms exactly the same - you can do a full text search for `resumable` to find relevant parts in multiple projects for instance.